frontend/src/metabase/palette/components/PaletteResultItem.tsx

```diff
@@ -35,22 +35,29 @@ export const PaletteResultItem = ({ item, active }: PaletteResultItemProps) => {
       c={active ? color("white") : color("text-dark")}
       aria-label={item.name}
     >
-      <Flex gap=".5rem" style={{ minWidth: 0 }}>
-        {item.icon && (
-          <Icon
-            aria-hidden
-            name={(item.icon as IconName) || "click"}
-            color={iconColor}
-            style={{
-              flexBasis: "16px",
-            }}
-          />
-        )}
+      {/** Icon Container */}
+      {item.icon && (
+        <Icon
+          aria-hidden
+          name={(item.icon as IconName) || "click"}
```

Comment:
Thinking out loud, maybe it would be useful to have a type guard so we can do: `isValidIconName(item.icon) ? item.icon : "click"`

Reply:
I actually just changed the typing a bit to remove this casting. (We cast somewhere else, but it's because the library doesn't allow us to tell it a type).